pkg/providers/common/common.go

```diff
@@ -74,12 +75,12 @@ func GenerateSSHAuthKey(writer filewriter.FileWriter) (string, error) {
 
 func CPMachineTemplateName(clusterName string, now types.NowFunc) string {
```

Comment:
we can define a new method

```go
CPMachineTemplateBase(clusterName string) string {
  return fmt.Sprintf("%s-control-plane-template", clusterName)
}
```

`CPMachineTemplateName` can use this method, the controller can also use it by `strings.HasPrefix(name, CPMachineTemplateBase)`

Same with Etcd. Preferably we should update worker name as well to follow same pattern.